### PR TITLE
chore: bumpenv pin numpy to 1.x [MD-470]

### PR DESCRIPTION
**Pull request by jgongd**
## Ticket

MD-470

## Description

The tensorflow in pytorch-tensorflow-cpu/cuda images is not compatible with numpy 2.x. Need to pin numpy to 1.x in these images. Stacktrace can be found in the ticket description.

Related environment PR: https://github.com/determined-ai/environments/pull/273

## Test Plan

Submit two experiments and see if they run successfully
1. Set cpu image to `determinedai/pytorch-tensorflow-cpu-dev:5432424` and make sure the experiment runs in a CPU resource pool.
2. Set cuda image to `determinedai/pytorch-tensorflow-cude-dev:5432424` and make sure the experiment runs in a GPU resource pool.

Sample config:
```
environment:
  image:
    cpu: determinedai/pytorch-tensorflow-cpu-dev:5432424
    cuda: determinedai/pytorch-tensorflow-cuda-dev:5432424
resources:
  resource_pool: default-cpu-pool
```

## Checklist

- [ ] Changes have been manually QA'd
- [ ] New features have been approved by the corresponding PM
- [ ] User-facing API changes have the "User-facing API Change" label
- [ ] Release notes have been added as a separate file under `docs/release-notes/`
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses have been included for new code which was copied and/or modified from any external code

[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by keita-determined — APPROVED** [review]

**Review by MikhailKardash — APPROVED** [review]
LGTM, just fix rebase issues
